horovod/spark/torch/remote.py

```diff
@@ -119,7 +121,14 @@ def train(serialized_model, optimizer_cls, model_opt_state_serialized,
         cuda_available = torch.cuda.is_available()
         if cuda_available:
             # Horovod: pin GPU to local rank.
-            torch.cuda.set_device(hvd.local_rank())
+            if os.environ.get('CUDA_VISIBLE_DEVICES') is None:
+                assert not is_dbfs
+                torch.cuda.set_device(hvd.local_rank())
+            else:
+                # Databricks pyspark sets CUDA_VISIBLE_DEVICES for GPU scheduling.
+                # Pin the only visible GPU allocated to this task.
+                assert is_dbfs
```

Comment:
This will fail in CI. We should use is_dbfs to determine whether to use pin to GPU[0]